api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -13,33 +14,68 @@ module OpenTelemetry.Internal.Logging.Types (
   severityInt,
 ) where
 
+import Control.Concurrent.Async
 import qualified Data.HashMap.Strict as H
 import Data.IORef (IORef)
 import Data.Int (Int64)
 import Data.Text (Text)
+import Data.Vector (Vector)
 import OpenTelemetry.Common (Timestamp, TraceFlags)
 import OpenTelemetry.Context.Types (Context)
-import OpenTelemetry.Internal.Common.Types (InstrumentationLibrary)
+import OpenTelemetry.Internal.Common.Types (InstrumentationLibrary, ShutdownResult)
 import OpenTelemetry.Internal.Trace.Id (SpanId, TraceId)
 import OpenTelemetry.LogAttributes
 import OpenTelemetry.Resource (MaterializedResources)
 
 
+data LogProcessor body = LogProcessor
```

Comment:
I think this should be called `LogRecordProcessor`: https://opentelemetry.io/docs/specs/otel/logs/sdk/#logrecordprocessor